34. Find First and Last Position of Element in Sorted Array/solution2.py

```diff
@@ -0,0 +1,42 @@
+"""
+70 ms
+14.4 MB
+TC : O(logN)
```

Comment:
The time complexity of your code is not `logN`.
## Explanation:
On line number 29 there is a nested while loop
```python
while nums[m]==target:
       m=m+1
       if m>=len(nums):
             break
```
Imagine the array size is 8 and your array is [1, 2, 3, 4, 4, 4, 4, 4]. 

Firstly the mid will be 7/2 i.e 3 so `nums[mid]` will be 4 now the inner for loop will run while `nums[m]==target`. So, it might run for the entire array. Making approx n/2   computations. so TC will be O(n/2) i.e 0(n) ultimately.

```suggestion
TC : O(N)
```